sql/keys.go

```diff
@@ -19,15 +19,52 @@ package sql
 
 import (
 	"strings"
+	"unicode"
+
+	"golang.org/x/text/unicode/norm"
 
 	"github.com/cockroachdb/cockroach/keys"
 	"github.com/cockroachdb/cockroach/proto"
 	"github.com/cockroachdb/cockroach/util/encoding"
 )
 
+// Normalize rules for Turkish/Azeri lowercase dotless-i and uppercase
+// dotted-i.
```

Comment:
This comment should explain what we're doing here and why: "Fold both dotted and dotless 'i' into the ascii i/I, so our case-insensitive comparison functions can be locale-invariant. This mapping implements case-insensitivity for Turkish and other latin-derived languages simultaneously, with the additional quirk that it is also insensitive to the dottedness of the i's."